src/util/segmentation/operations/correction.js

```diff
@@ -0,0 +1,524 @@
+import { fillInside, eraseInside } from './index.js';
+import getPixelPathBetweenPixels from './getPixelPathBetweenPixels';
+import clip from '../../clip';
+import pointInImage from '../../pointInImage';
+
+import { getLogger } from '../../logger';
+import floodFill from './floodFill.js';
+
+const logger = getLogger('util:segmentation:operations:correction');
+
+/**
+ * correction - Using the stroke given, determine which action(s) to perfom:
+ * - Stroke starts and ends inside a segmentation: Behaves as an subtractive freehand scissors.
+ * - Stroke starts and ends outside a segmentation: Behaves as an additive freehand scissors.
+ * - Stroke out-in-out: Section is subtracted.
+ * - Stroke in-out-in: Section is added.
+ *
+ * @param  {Object} evt The cornerstone event.
```

Comment:
`evt` is a cornerstone event, but it contains `operationData`?

Reply:
Yeah, this is the mechanism Romulo and Rodrigo came up with as a way to use the `applyStrategy` method in base tool instead of reinventing the wheel for different scissor modes. As the segmentation tools have grown a bit, perhaps we should formalise this. `applyStrategy` should just be able to pass an additional argument. I'll refactor this.

I can do it in a non-breaking way so its not another thing on the list for migration docs.

Reply:
Done.

Reply:
I'll keep my eyes open for the implementation in changes ^_^